data/transactions/logic/assembler_test.go

```diff
@@ -267,141 +267,54 @@ pop
 	require.Equal(t, prog1, prog2)
 }
 
+func testLine(t *testing.T, line string, ver uint64, expected string) {
+	// By embedding the source line between two other lines, the
+	// test for the correct line number in the error is more
+	// meaningful.
+	source := "int 1\n" + line + "\nint 1\n"
+	_, err := AssembleStringWithVersion(source, ver)
+	if expected == "" {
+		require.NoError(t, err)
+	} else {
+		require.Error(t, err)
+		if strings.HasSuffix(expected, "...") {
+			require.Contains(t, "^"+err.Error(), "^2: "+expected[:len(expected)-3])
```

Comment:
Is `^` really needed?

Reply:
Again, I'm making the test a bit harder to pass and a more "real" test.  It's conceivable that an error could report line 12 instead of line 2, and without the extra delimiter, Contains() would still report success because the message started with "2".